components/director/internal/model/application.go

```diff
@@ -17,11 +17,17 @@ type Application struct {
 	IntegrationSystemID *string
 }
 
-func (app *Application) SetFromUpdateInput(update ApplicationUpdateInput) {
+func (app *Application) SetFromUpdateInput(update ApplicationUpdateInput, timestamp time.Time) {
+	if app.Status == nil {
+		app.Status = &ApplicationStatus{}
+	}
+
 	app.Description = update.Description
 	app.HealthCheckURL = update.HealthCheckURL
 	app.IntegrationSystemID = update.IntegrationSystemID
 	app.ProviderName = update.ProviderName
+	app.Status.Condition = getApplicationStatusConditionOrDefault(update.StatusCondition)
```

Comment:
When we update application name and other properties, do we should reset `Condition` of Application?